_posts/2018-06-26-viroreact.md

```diff
@@ -0,0 +1,643 @@
+
+For this blogpost I will only discuss how we created the ViroReact application.
+
+### The AR Meeting room viewer (ViroReact)
+
+<a href="{{ '/img/2018-06-26-viroreact/app-scan.png' | prepend: site.baseurl }}" data-lightbox="app-scan" data-title="meeting-room-viewer-app-foto">
+    <img alt="stack" src="{{ '/img/2018-06-26-viroreact/app-scan.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 500px;">
+</a>
+
+ViroReact is built on top of React, so basic knowledge of React is necessary. 
+To get started, you can follow this great free beginner guide by Kent C. Dodds at egghead.io 
+[The beginner's guide to React](https://egghead.io/courses/the-beginner-s-guide-to-react.)
+
+
+#### Step 1: Project setup
+We began by following the Official [Quick Start guide from ViroMedia](https://docs.viromedia.com/v2.7.3/docs/quick-start).
+
+Using the [react-viro CLI](https://www.npmjs.com/package/react-viro), we generated a ViroSample project.
+
+```bash
+react-viro init ARMeetingRoomViewer
+``` 
+
+#### Step 2: Create an AR scene for viewing the status of a meeting room.
+To be able to scan the meeting room we needed a picture of the meeting room nameplate, 
+this will act as the marker to scan the meeting room information.
+
+<a href="{{ '/img/2018-06-26-viroreact/m07-image-marker.png' | prepend: site.baseurl }}" data-lightbox="m07" data-title="meeting-room-viewer-app-foto">
+    <img alt="stack" src="{{ '/img/2018-06-26-viroreact/m07-image-marker.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 300px;">
+</a>
+
+I placed the marker image inside of the `**/js/res**` folder.
+
+Because the file extension was in capital letters (.JPG), 
+I had to configure this extension in the `rn-cli.config.js` file inside of the root folder.
+
+```javascript
+  getAssetExts() {
+    return ["obj", "mtl", "JPG", "vrx", "hdr"];
+  },
+```
+
+Next, I created the actual scene in a file called **markerscene.js** in the **/js/** folder.
+
+To be able to scan the image marker, we need 2 important APIs:
+- [ViroARTrackingTargets](https://docs.viromedia.com/v2.7.3/docs/viroartrackingtargets)
+- [ViroARImageMarker](https://docs.viromedia.com/v2.7.3/docs/viroarimagemarker) component
+
+When the scene initialises we need to setup the Tracking Target(our image marker)
+**mr7** refers to a meeting room name.
+
+We call this method inside of the constructor.
+
+```javascript
+    setMarker() {
+        ViroARTrackingTargets.createTargets({
+            "mr7": {
+                source: require('./res/mr7.JPG'),
+                orientation: "Up",
+                physicalWidth: 1
+            },
+        });
+    }
+```
+
+Then we need to define what to render. 
+The root component will be an AR Scene:
+
+```xml
+ <ViroARScene onAnchorFound={this._getInfo}
+              onClick={this._getInfo}>
+  ...
+ </ViroARScene>
+```
+
+We bind 2 events on the ViroARScene component, onAnchorFound and onClick. 
```

Comment:
We bind two events on the ViroARScene component: `onAnchorFound` and `onClick`.(Two, colon and backticks)